cmd/routed-eni-cni-plugin/driver/driver.go

```diff
@@ -282,6 +252,151 @@ func setupNS(hostVethName string, contVethName string, netnsPath string, addr *n
+	ns nswrapper.NS, mtu int, procSys procsyswrapper.ProcSys, log logger.Logger) (netlink.Link, error) {
+	// Clean up if hostVeth exists.
+	if oldHostVeth, err := netLink.LinkByName(hostVethName); err == nil {
+		if err = netLink.LinkDel(oldHostVeth); err != nil {
+			return nil, errors.Wrapf(err, "setupVeth network: failed to delete old hostVeth %q", hostVethName)
+		}
+		log.Debugf("Cleaned up old hostVeth: %v\n", hostVethName)
+	}
+
+	createVethContext := newCreateVethPairContext(contVethName, hostVethName, addr, mtu)
+	if err := ns.WithNetNSPath(netnsPath, createVethContext.run); err != nil {
+		log.Errorf("Failed to setup veth network %v", err)
+		return nil, errors.Wrap(err, "setupVeth network: failed to setup veth network")
+	}
+
+	hostVeth, err := netLink.LinkByName(hostVethName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "setupVeth network: failed to find link %q", hostVethName)
+	}
+
+	// NB: Must be set after move to host namespace, or kernel will reset to defaults.
+	if err := procSys.Set(fmt.Sprintf("net/ipv6/conf/%s/accept_ra", hostVethName), "0"); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, errors.Wrapf(err, "setupVeth network: failed to disable IPv6 router advertisements")
+		}
+		log.Debugf("setupVeth network: Ignoring '%s' writing to accept_ra: Assuming kernel lacks IPv6 support", err)
+	}
+
+	if err := procSys.Set(fmt.Sprintf("net/ipv6/conf/%s/accept_redirects", hostVethName), "0"); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, errors.Wrapf(err, "setupVeth network: failed to disable IPv6 ICMP redirects")
+		}
+		log.Debugf("setupVeth network: Ignoring '%s' writing to accept_redirects: Assuming kernel lacks IPv6 support", err)
+	}
+	log.Debugf("setupVeth network: disabled IPv6 RA and ICMP redirects on %s", hostVethName)
+
+	// Explicitly set the veth to UP state, because netlink doesn't always do that on all the platforms with net.FlagUp.
+	// veth won't get a link local address unless it's set to UP state.
+	if err = netLink.LinkSetUp(hostVeth); err != nil {
+		return nil, errors.Wrapf(err, "setupVeth network: failed to set link %q up", hostVethName)
+	}
+	return hostVeth, nil
+}
+
+// SetupPodENINetwork sets up the network ns for pods requesting its own security group
+func (os *linuxNetwork) SetupPodENINetwork(hostVethName string, contVethName string, netnsPath string, addr *net.IPNet,
+	vlanId int, eniMAC string, subnetGW string, parentIfIndex int, mtu int, log logger.Logger) error {
+
+	hostVeth, err := setupVeth(hostVethName, contVethName, netnsPath, addr, os.netLink, os.ns, mtu, os.procSys, log)
+	if err != nil {
+		return errors.Wrapf(err, "SetupPodENINetwork failed to setup veth pair.")
+	}
+
+	vlanTableId := vlanId + 100
+	vlanLink := buildVlanLink(vlanId, parentIfIndex, eniMAC)
+
+	// 1. clean up if vlan already exists (necessary when trunk ENI changes).
+	if oldVlan, err := os.netLink.LinkByName(vlanLink.Name); err == nil {
+		if err = os.netLink.LinkDel(oldVlan); err != nil {
```

Comment:
If err != nil it will proceed further. Do you mean to say proceed only if err != nil and is`NotFound`? I can add but thinking it might not be necessary.